db/migrate/20220809152005_fix_ibm_power_vc_sti_classes.rb

```diff
@@ -0,0 +1,168 @@
+    self.inheritance_column = :_type_disabled
+
+    belongs_to :ext_management_system, :foreign_key => :ems_id, :class_name => "FixIbmPowerVcStiClasses::ExtManagementSystem"
+  end
+
+  class CloudSubnet < ActiveRecord::Base
+    include ActiveRecord::IdRegions
+    self.inheritance_column = :_type_disabled
+
+    belongs_to :ext_management_system, :foreign_key => :ems_id, :class_name => "FixIbmPowerVcStiClasses::ExtManagementSystem"
+  end
+
+  class FloatingIp < ActiveRecord::Base
+    include ActiveRecord::IdRegions
+    self.inheritance_column = :_type_disabled
+
+    belongs_to :ext_management_system, :foreign_key => :ems_id, :class_name => "FixIbmPowerVcStiClasses::ExtManagementSystem"
+  end
+
+  class NetworkPort < ActiveRecord::Base
+    include ActiveRecord::IdRegions
+    self.inheritance_column = :_type_disabled
+
+    belongs_to :ext_management_system, :foreign_key => :ems_id, :class_name => "FixIbmPowerVcStiClasses::ExtManagementSystem"
+  end
+
+  class NetworkRouter < ActiveRecord::Base
+    include ActiveRecord::IdRegions
+    self.inheritance_column = :_type_disabled
+
+    belongs_to :ext_management_system, :foreign_key => :ems_id, :class_name => "FixIbmPowerVcStiClasses::ExtManagementSystem"
+  end
+
+  class SecurityGroup < ActiveRecord::Base
+    include ActiveRecord::IdRegions
+    self.inheritance_column = :_type_disabled
+
+    belongs_to :ext_management_system, :foreign_key => :ems_id, :class_name => "FixIbmPowerVcStiClasses::ExtManagementSystem"
+  end
+
+  class CloudVolume < ActiveRecord::Base
+    include ActiveRecord::IdRegions
+    self.inheritance_column = :_type_disabled
+
+    belongs_to :ext_management_system, :foreign_key => :ems_id, :class_name => "FixIbmPowerVcStiClasses::ExtManagementSystem"
+  end
+
+  class CloudVolumeBackup < ActiveRecord::Base
+    include ActiveRecord::IdRegions
+    self.inheritance_column = :_type_disabled
+
+    belongs_to :ext_management_system, :foreign_key => :ems_id, :class_name => "FixIbmPowerVcStiClasses::ExtManagementSystem"
+  end
+
+  class CloudVolumeSnapshot < ActiveRecord::Base
+    include ActiveRecord::IdRegions
+    self.inheritance_column = :_type_disabled
+
+    belongs_to :ext_management_system, :foreign_key => :ems_id, :class_name => "FixIbmPowerVcStiClasses::ExtManagementSystem"
+  end
+
+  class CloudVolumeType < ActiveRecord::Base
+    include ActiveRecord::IdRegions
+    self.inheritance_column = :_type_disabled
+
+    belongs_to :ext_management_system, :foreign_key => :ems_id, :class_name => "FixIbmPowerVcStiClasses::ExtManagementSystem"
+  end
+
+  def up
+    say_with_time("Fixing IBM PowerVC Network STI classes") do
+      network_klass = "ManageIQ::Providers::IbmPowerVc::NetworkManager"
+
+      network_managers = ExtManagementSystem.in_my_region.where(:type => network_klass)
+
+      CloudNetwork.in_my_region
+                  .where(:ext_management_system => network_managers)
+                  .update_all(:type => "#{network_klass}::CloudNetwork")
```

Comment:
@jaywcarman were the types `nil` before or were they e.g. `Openstack::NetworkManager::CloudNetwork...`?

Wondering because of `ManageIQ::Providers::IbmPowerVc::NetworkManager::CloudNetwork::Private` and `ManageIQ::Providers::IbmPowerVc::NetworkManager::CloudNetwork::Public` are the classes which will be actually set

Reply:
I think the types were all set, just in the wrong namespace.

The parent ems types are correctly in `ManageIQ::Providers::IbmPowerVc::`, though, so I was trying to select via `.where(:ext_management_system => network_managers)`.

Reply:
Oh yeah that is fine, it is just for the cloud networks where the types are actually going to be `#{network_klass}::CloudNetwork::Public"` or `#{network_klass}::CloudNetwork::Private"` could we do:

```
      CloudNetwork.in_my_region
                  .where(:ext_management_system => network_managers)
                  .where(:type => "ManageIQ::Providers::Openstack::NetworkManager::CloudNetwork::Public")
                  .update_all(::type => "#{network_klass}::CloudNetwork::Public")
```

Reply:
(If they were `nil` we'd have no idea what they _should_ be)

Reply:
I just forced pushed an update with the Public and Private CloudNetwork types included.